lib/hurley/url.rb

```diff
@@ -152,7 +152,7 @@ def raw_query=(new_query)
     def basic_auth
       return unless @user
       userinfo = @password ? "#{@user}:#{@password}" : @user
-      "Basic #{Base64.encode64(userinfo).rstrip}"
+      "Basic #{Base64.encode64(userinfo).rstrip.gsub(/\n/, '')}"
```

Comment:
We can probably leave the `rstrip` out.  But I'd make this change after replacing the test below with the expected output, and not the code to generate it.